### PR TITLE
#164798297 Users can follow each other

### PR DESCRIPTION
**Pull request by Genza999**
**What does this PR do?**
Creates functionality for users following each other

**Description of tasks to be completed.**

- Creation of functionality of users to follow each other

- Creation of functionality of users to unfollow each other

**How should this be manually tested?**

- After cloning the repo, cd into it and run the following commands to activate the virtual environment
   `virtualenv venv`
   `source venv/bin/activate`

- Checkout the branch with the functionality of users folllowing/unfollowing eachother
   `git checkout ft-users-follow-each-other-164798297`

- Install all requirements
   `pip install -r requirements.txt`

- Source the environment variables
   `source .env`

- Run the migrations
   `python manage.py makemigrations`
   `python manage.py migrate`

- Run the server
   `python manage.py runserver`

- To test out this functionality one has to first register on the application, in case of a successful registration, they will receive an account activation link on their email, this activation link would then be pasted in the URL section of postman and sent in order to activate and log into ones account where they would get a token which they would use in the Authorization section to access all other routes including the follows route.

- To follow a user, one has to send a **POST** request to the route:
  `http://127.0.0.1:8000/api/v1/profiles/{username}/follow` (see Figure 1)

Screen Shot 2019-04-10 at 10 22 42

**Figure 1** shows successful following of a user anyati after which the response _You are now following anyati_ is returned.

- To unfollow a user, one has to send a **DELETE** request to the route:
   `http://127.0.0.1:8000/api/v1/profiles/{username}/follow` (see Figure 2)

Screen Shot 2019-04-10 at 10 27 11

**Figure 2** shows successful unfollowing of a user anyati after which the response _You have unfollowed anyati_ is returned.

**What are the relevant pivotal tracker stories?**
   [#164798297](https://www.pivotaltracker.com/story/show/164798297)

### CODE REVIEW

**Review by patrickf949 — APPROVED** [review]
@Genza999 Neat Implementation

**Review by e-ian — APPROVED** [review]
Great work on readability of the code and descriptive function names. LGTM

**Review by anyatibrian — APPROVED** [review]
great work, really good implementation

File: authors/apps/profiles/models.py [hunk 1]
```diff
@@ -13,10 +13,33 @@ class creating a data model to hold information about users
     bio = models.TextField(blank=True)
     full_name = models.CharField(max_length=255, blank=True)
     image = models.URLField(blank=True)
+    follows = models.ManyToManyField(
+        'self',
+        related_name='followed_by',
+        symmetrical=False
+    )
+    followers_no = models.PositiveIntegerField(default=0)
```
Comment (inno-asiimwe) [inline authors/apps/profiles/models.py]: We do not need to store these numbers in the database, they can always be calculated when needed.

Comment (Genza999) [reply authors/apps/profiles/models.py]: Thanks, I have worked on it! I have used SerializerMethod as suggested. It is Perfect Now!

**Review by patrickf949 — APPROVED** [review]

**Review by inno-asiimwe — APPROVED** [review]
LGTM

**Pull request merged**